### PR TITLE
Refactor admin audit app

### PR DESCRIPTION
**Pull request by danialRahimy**
## Summary
Refactor admin audit app

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/admin_audit/lib/AppInfo/Application.php [hunk 1]
```diff
@@ -70,9 +71,7 @@
 use Psr\Log\LoggerInterface;
 
 class Application extends App implements IBootstrap {
-
-	/** @var LoggerInterface */
-	protected $logger;
+	protected LoggerInterface $logger;
```
[the remark this answers is not in the document]
Comment (danialRahimy) [reply apps/admin_audit/lib/AppInfo/Application.php]: `$logger` property is defined in the boot methods, I think it is a convention. Do I need to define it in the constructor?

File: apps/admin_audit/lib/Actions/Action.php [hunk 2]
```diff
@@ -49,23 +49,26 @@
...
+			if ($obfuscateParameters) {
+				$this->logger->critical(
+					'$params["'.$element.'"] was missing.',
+					['app' => 'admin_audit']
+				);
+			} else {
+				$this->logger->critical(
+					sprintf(
+						'$params["'.$element.'"] was missing. Transferred value: %s',
+						print_r($params, true)
```
[the remark this answers is not in the document]
Comment (danialRahimy) [reply apps/admin_audit/lib/Actions/Action.php:65]: According to the second param of print_r that is `true`, there is no output and XSS vulnerability.

**Comment by danialRahimy** [comment]
Hey @szaimen

I opened this PR last week, but haven't been assigned to any reviewers yet. Is there anything that I could do?

Thanks.

**Comment by szaimen** [comment]
Thanks for the ping! I requested some reviewers.

**Comment by danialRahimy** [comment]
Dear @szaimen

Would it be necessary for me to take any action in order to proceed with the pull request?

**Comment by szaimen** [comment]
@nextcloud/server-backend please review!

File: apps/admin_audit/lib/AuditLogger.php [hunk 3]
```diff
@@ -50,39 +48,39 @@ public function __construct(ILogFactory $logFactory, IConfig $config) {
 		$this->parentLogger = $logFactory->getCustomPsrLogger($logFile, $auditType, $auditTag);
 	}
 
-	public function emergency($message, array $context = array()) {
+	public function emergency($message, array $context = array()): void {
```
Comment (Altahrim) [inline apps/admin_audit/lib/AuditLogger.php:51]: I would prefer the new way:
```suggestion
	public function emergency($message, array $context = []): void {
```
Same for next functions

File: apps/admin_audit/lib/Actions/Sharing.php [hunk 4]
```diff
@@ -44,136 +44,168 @@ class Sharing extends Action {
...
+			IShare::TYPE_USER => $this->sharedUserType($params),
+			IShare::TYPE_GROUP => $this->sharedGroupType($params),
+			IShare::TYPE_ROOM => $this->sharedRoomType($params),
+			IShare::TYPE_EMAIL => $this->sharedEmailType($params),
+			IShare::TYPE_CIRCLE => $this->sharedCircleType($params),
+			IShare::TYPE_REMOTE => $this->sharedRemoteType($params),
+			IShare::TYPE_REMOTE_GROUP => $this->sharedRemoteGroupType($params),
+			IShare::TYPE_DECK => $this->sharedDeckType($params),
+			IShare::TYPE_SCIENCEMESH => $this->sharedSciencemeshType($params),
+			default => null,
```
Comment (Altahrim) [inline apps/admin_audit/lib/Actions/Sharing.php:58]: I wonder if this one should fail (or at least log) if an unknown type is used.

**Review by Altahrim — APPROVED** [review]

File: apps/admin_audit/lib/Actions/Sharing.php (hunk 4 above)
Comment (Altahrim) [reply apps/admin_audit/lib/Actions/Sharing.php:58]: Same here, maybe it will ring a bell for someone :)

**Review by fsamapoor — COMMENTED** [review]
In my opinion, removing redundant annotations make the code that much cleaner. When a method's parameter is typed, having the exact type hint as an annotation does not add any value to the mix. I think it would be great if you could also review the annotations in this PR.

I would also recommend writing a summary of the changes you have made in your PR's description and explaining briefly how those changes would improve the codebase.

File: apps/admin_audit/lib/Actions/Sharing.php [hunk 5]
```diff
@@ -44,136 +44,168 @@ class Sharing extends Action {
...
-					'itemType',
-					'path',
-					'itemSource',
-					'shareWith',
-					'permissions',
-					'id',
-				]
-			);
-		}
+		match ($params['shareType']) {
```
Comment (come-nc) [inline apps/admin_audit/lib/Actions/Sharing.php:47]: The use of `match` when the return value is not used is really confusing.

File: apps/admin_audit/lib/AppInfo/Application.php [hunk 6]
```diff
@@ -60,6 +60,7 @@
 use OCP\EventDispatcher\IEventDispatcher;
 use OCP\IConfig;
 use OCP\IGroupManager;
+use OCP\IServerContainer;
```
Comment (come-nc) [inline apps/admin_audit/lib/AppInfo/Application.php:63]: Why? It does not seem used.

**Review by come-nc — COMMENTED** [review]
This is only moving things around, and is using `match` in a weird way.
I’m not sure I get the point.

**Comment by skjnldsv** [comment]
Hey @danialRahimy  thanks for the PR
Unfortunately there seem to be quite a bit of conflicts and some comments left unanswered above, so we closed that one. If you want to re-open and ork on it again, feel free to do so and ping me, we'll be happy to help you and get it move forward

If you used a tooling to create those, feel free to run it again now and ping me @skjnldsv straight away! I'll help you get it merged faster :rocket:

Have a nice day :)

**Pull request closed without merging**